internal/controller/certificates/policies/checks.go

```diff
@@ -347,134 +473,159 @@ func SecretTemplateMismatchesSecretManagedFields(fieldManager string) Func {
 			}
 		}
 
-		baseAnnotations, err := internalcertificates.AnnotationsForCertificate(x509cert)
+		expCertificateDataAnnotations, err := internalcertificates.AnnotationsForCertificate(x509cert)
```

Comment:
This `SecretManagedLabelsAndAnnotationsManagedFieldsMismatch` function makes sure the Secret's certificate details annotations all exist. It uses the output of `AnnotationsForCertificate` to build the set of expected annotations. NOTE: this function does not check the contents of these annotations, that happens in `SecretCertificateDetailsAnnotationsMismatch`.